runtime/src/utils/account_id.rs

```diff
@@ -0,0 +1,176 @@
+use crate::SS58_PREFIX;
+use base58::{FromBase58, ToBase58};
+use blake2::{Blake2b512, Digest};
+use codec::{Decode, Encode};
+use serde::{Deserialize, Serialize};
+use std::convert::TryInto;
+/// A 32-byte cryptographic identifier. This is a simplified version of Substrate's
+/// `sp_core::crypto::AccountId32`. To obtain more functionality, convert this into
+/// that type.
+#[derive(
+    Hash,
+    Clone,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Encode,
+    Decode,
+    Debug,
+    scale_encode::EncodeAsType,
+    scale_decode::DecodeAsType,
+    Default,
+)]
+pub struct AccountId32(pub [u8; 32]);
+
+impl AsRef<[u8]> for AccountId32 {
+    fn as_ref(&self) -> &[u8] {
+        &self.0[..]
+    }
+}
+
+impl AsRef<[u8; 32]> for AccountId32 {
+    fn as_ref(&self) -> &[u8; 32] {
+        &self.0
+    }
+}
+
+impl From<[u8; 32]> for AccountId32 {
+    fn from(x: [u8; 32]) -> Self {
+        AccountId32(x)
+    }
+}
+
+impl AccountId32 {
+    pub fn new(value: [u8; 32]) -> Self {
+        AccountId32(value)
+    }
+
+    // Return the ss58-check string for this key. Adapted from `sp_core::crypto`. We need this to
+    // serialize our account appropriately but otherwise don't care.
+    pub fn to_ss58check(&self) -> String {
+        // For serializing to a string to obtain the account nonce, we use the default substrate
+        // prefix (since we have no way to otherwise pick one). It doesn't really matter, since when
+        // it's deserialized back in system_accountNextIndex, we ignore this (so long as it's valid).
+        // prefix <= 63 just take up one byte at the start:
+        let mut v = vec![SS58_PREFIX as u8];
```

Comment:
Can you please add a test? It seems to me like this shouldn't work - `SS58_PREFIX` is either 2032 or 2092, both of which don't fit into u8. Please also add permalinks to where you copied this from.

See https://github.com/paritytech/substrate/blob/696d09b3d1da1b2045fe46200df1d9f6d6d9d829/primitives/core/src/crypto.rs#L328-L340 for how to  use prefixes > 256

Reply:
I have referenced it from [subxt utils](https://github.com/paritytech/subxt/tree/master/subxt/src/utils). Do you want me to modify it according to `sp_core` reference provided above or depend on subxt itself?

Reply:
Modified implementation according to `sp_core`